proto/frequenz/api/common/v1/sensors.proto

```diff
@@ -76,3 +78,102 @@ enum SensorMetric {
+//   Example output of a component data message:
+//   ```
+//    {
+//      component_id: 13,
+//      metric_samples: [
+//        /* list of metrics for multiple timestamps */
+//        {
+//          sampled_at: "2023-10-01T00:00:00Z",
+//          metric: "METRIC_SENSOR_TEMPERATURE",
+//          sample: metric_sample_type: {simple_metric: {value: 23.5},
+//          bounds: {},
+//        },
+//        {
+//          sampled_at: "2023-10-01T00:00:00Z",
+//          metric: "METRIC_SENSOR_RELATIVE_HUMIDITY",
+//          sample: metric_sample_type: {simple_metric: {value: 23.5},
+//          bounds: {},
+//        }
+//      ]
+//    }
+//  ```
+message SensorData {
+  // The ID of the microgrid sensors.
+  uint64 sensor_id = 1;
+
+  // List of measurements for a metric of the specific microgrid sensor.
+  repeated SensorMetricSample metric_samples = 2;
+
+  // List of states of a specific microgrid sensor.
+  repeated SensorState states = 3;
+}
+
+// Representation of a sensor state and errors.
+message SensorState {
+  // The time at which the state was sampled.
+  google.protobuf.Timestamp sampled_at = 1;
+
+  // List of states of the microgrid sensor.
+  //
+  // !!! note
+  //    The list will contain unique members. No state will exist twice in
+  //    this list.
+  repeated SensorStateCode states = 2;
+
+  // List of errors for the microgrid sensor.
+  //
+  // !!! note
+  //    This list is expected to have errors if and only if the sensor is in
+  //    an error state.
+  //
+  // !!! note
+  //    The list will contain unique members. No error will exist twice in
+  //    this list.
+  repeated SensorErrorCode errors = 4;
+}
+
+// Enum to represent the various states that a sensor can be in.
+// This enum is unified across all sensor categories for consistency.
+enum SensorStateCode {
+  // Default value when the sensor state is not explicitly set.
+  // This is the zero value of the enum.
+  SENSOR_STATE_CODE_UNSPECIFIED = 0;
+
+  // The sensor is up and running.
+  SENSOR_STATE_CODE_ON = 1;
+
+  // The sensor is in an error state.
+  SENSOR_STATE_CODE_ERROR = 2;
+}
+
+// A representation of all possible errors that can occur in sensors.
+enum SensorErrorCode {
+  // Default value. No specific error is specified.
+  SENSOR_ERROR_CODE_UNSPECIFIED = 0;
+
+  // The sensor is reporting an unknown or an undefined error, and the sender
+  // cannot parse the sensor error to any of the variants below.
+  SENSOR_ERROR_CODE_UNKNOWN = 1;
+
+  // Error indicating an internal error within the sensor.
+  SENSOR_ERROR_CODE_INTERNAL = 2;
+}
+
+// Representation of a sampled sensor metric along with its value.
+message SensorMetricSample {
+  // The UTC timestamp of when the metric was sampled.
+  google.protobuf.Timestamp sampled_at = 1;
+
+  // The metric that was sampled.
+  frequenz.api.common.v1.metrics.Metric metric = 2;
+
+  // The value of the sampled metric.
+  frequenz.api.common.v1.metrics.MetricSampleVariant sample = 3;
+}
```

Comment:
This entire message is just a copy paste of `metrics.MetricSample`, but bounds deleted. Does it make sense to copy it like this? In the future, if we change `metrics.MetricSample`, then we also would need to change this message here. Alternatively, we could just use `metrics.MetricSample` instead of this one, but never populate the bounds for sensors.

Reply:
> Alternatively, we could just use metrics.MetricSample instead of this one, but never populate the bounds for sensors.

No I don't feel thats a good idea.

> Does it make sense to copy it like this?

I guess so until we know more about what sensors we are going to support. My assumption is that maybe we will replace `MetricSampleVariant` with its own version just for sensors.